core/string/ustring.cpp

```diff
@@ -1636,19 +1636,21 @@ String String::num_real(double p_num, bool p_trailing) {
 
 	bool neg = p_num < 0;
 	p_num = ABS(p_num);
-	int intn = (int)p_num;
+	int64_t intn = (int64_t)p_num;
 
 	// Decimal part.
 
 	if (intn != p_num) {
-		double dec = p_num - (double)(intn);
+		double dec = p_num - (double)intn;
 
 		int digit = 0;
 
 #if REAL_T_IS_DOUBLE
```

Comment:
By the way, should this be `#ifdef`? I'm not sure which is preferred. Both seem to work, but `#ifdef` is used elsewhere in the C++ code.

Reply:
Yes. That seems to be the only occurrence outside C# files, so I'd suggest fixing it in this PR.